homeassistant/components/miele/const.py

```diff
@@ -167,174 +167,272 @@ class MieleActions(IntEnum):
     "stop_supercooling": MieleActions.STOP_SUPERCOOL,
 }
 
-STATE_PROGRAM_PHASE_WASHING_MACHINE = {
-    0: "not_running",  # Returned by the API when the machine is switched off entirely.
-    256: "not_running",
-    257: "pre_wash",
-    258: "soak",
-    259: "pre_wash",
-    260: "main_wash",
-    261: "rinse",
-    262: "rinse_hold",
-    263: "cleaning",
-    264: "cooling_down",
-    265: "drain",
-    266: "spin",
-    267: "anti_crease",
-    268: "finished",
-    269: "venting",
-    270: "starch_stop",
-    271: "freshen_up_and_moisten",
-    272: "steam_smoothing",
-    279: "hygiene",
-    280: "drying",
-    285: "disinfecting",
-    295: "steam_smoothing",
-    65535: "not_running",  # Seems to be default for some devices.
-}
 
-STATE_PROGRAM_PHASE_TUMBLE_DRYER = {
-    0: "not_running",
-    512: "not_running",
-    513: "program_running",
-    514: "drying",
-    515: "machine_iron",
-    516: "hand_iron_2",
-    517: "normal",
-    518: "normal_plus",
-    519: "cooling_down",
-    520: "hand_iron_1",
-    521: "anti_crease",
-    522: "finished",
-    523: "extra_dry",
-    524: "hand_iron",
-    526: "moisten",
-    527: "thermo_spin",
-    528: "timed_drying",
-    529: "warm_air",
-    530: "steam_smoothing",
-    531: "comfort_cooling",
-    532: "rinse_out_lint",
-    533: "rinses",
-    535: "not_running",
-    534: "smoothing",
-    536: "not_running",
-    537: "not_running",
-    538: "slightly_dry",
-    539: "safety_cooling",
-    65535: "not_running",
-}
+class ProgramPhaseWashingMachine(MieleEnum):
+    """Program phase codes for washing machines."""
 
-STATE_PROGRAM_PHASE_DISHWASHER = {
-    1792: "not_running",
-    1793: "reactivating",
-    1794: "pre_dishwash",
-    1795: "main_dishwash",
-    1796: "rinse",
-    1797: "interim_rinse",
-    1798: "final_rinse",
-    1799: "drying",
-    1800: "finished",
-    1801: "pre_dishwash",
-    65535: "not_running",
-}
+    not_running = 0, 256, 65535
+    pre_wash = 257, 259
+    soak = 258
+    main_wash = 260
+    rinse = 261
+    rinse_hold = 262
+    cleaning = 263
+    cooling_down = 264
+    drain = 265
+    spin = 266
+    anti_crease = 267
+    finished = 268
+    venting = 269
+    starch_stop = 270
+    freshen_up_and_moisten = 271
+    steam_smoothing = 272, 295
+    hygiene = 279
+    drying = 280
+    disinfecting = 285
+    missing2none = -9999
```

Comment:
I think this `missing2none` hack with a special value is quite ugly.
I'd suggest to instead use `homeassistant.util.enum.try_parse_enum`

Also, I'd strongly suggest to get rid of the warning prints in `MieleEnum`, errors should bubble up as exceptions to the caller who decides what to do with them (for example log).